### ISSUE TITLE
Main release files are not named with the same name as the signature file (and it is confusing)

### ISSUE THREAD

**Issue by sergioprg** [coreruleset/coreruleset#4211]
### Describe the bug

The GPG signature `.asc` file for release **v4.16.0** does not match the current `.tar.gz` tarball hosted on GitHub.

When verifying the release file with the official CRS key (`38EEACA1AB8A6E72`), GPG reports a **BAD signature**.

This may indicate that the `.tar.gz` was modified or regenerated after the `.asc` was created.

---

### Steps to reproduce

```bash
wget https://github.com/coreruleset/coreruleset/releases/download/v4.16.0/coreruleset-4.16.0.tar.gz
wget https://github.com/coreruleset/coreruleset/releases/download/v4.16.0/coreruleset-4.16.0.tar.gz.asc

gpg --keyserver hkps://keys.openpgp.org --recv-keys 38EEACA1AB8A6E72

gpg --verify coreruleset-4.16.0.tar.gz.asc coreruleset-4.16.0.tar.gz
```

---

### Expected behaviour

GPG should return a **valid signature** for the tarball using the `.asc` file and the official CRS team key.

---

### Actual behaviour

```plaintext
gpg: Signature made Sun Jun 29 11:51:17 2025 -03
gpg:                using RSA key 36006F0E0BA167832158821138EEACA1AB8A6E72
gpg:                issuer "security@coreruleset.org"
gpg: BAD signature from "OWASP Core Rule Set <security@coreruleset.org>" [unknown]
```

---

### Additional context

- The `.asc` file is structurally correct and references the official CRS key `38EEACA1AB8A6E72`.
- The `.tar.gz` content does not match the signature.
- The key used (`38EEACA1AB8A6E72`) is listed in the [official documentation](https://coreruleset.org/docs/1-getting-started/1-1-crs-installation/).
- ⚠️ Note: The GitHub release tag `v4.16.0` shows that it was signed by Felipe Zipitría using key `4FECAA9B18A621DE`, which differs from the key used to sign the `.asc` file. This could lead to confusion when validating the authenticity of the release.

---

### Suggested fix

- Re-upload the **original tarball** that was signed with this `.asc`, or  
- Generate a **new `.asc` signature** for the current `.tar.gz`.

---

### My Environment

_Not applicable – issue occurs before runtime during installation integrity checks._

---

🙏 **Thank you very much for your continued work on the CRS project and for dedicating your time to improving web application security and the broader Internet ecosystem.**  
This verification process is essential in high-security and automated environments, and we truly appreciate your attention to this matter.

**Comment by fzipi** [comment]
Ok, tried to repeat what you did there; there is a catch:

👉 `coreruleset-4.16.0.tar.gz` does not exist. 

You need to use:
- the default https://github.com/coreruleset/coreruleset/archive/refs/tags/v4.16.0.tar.gz, and use https://github.com/coreruleset/coreruleset/releases/download/v4.16.0/coreruleset-4.16.0.tar.gz.asc
- or the minimal release: https://github.com/coreruleset/coreruleset/releases/download/v4.16.0/coreruleset-4.16.0-minimal.tar.gz, and use https://github.com/coreruleset/coreruleset/releases/download/v4.16.0/coreruleset-4.16.0-minimal.tar.gz.asc

```
❯ ls
coreruleset-4.16.0.tar.gz.asc v4.16.0.tar.gz
❯ gpg --verify coreruleset-4.16.0.tar.gz.asc v4.16.0.tar.gz
gpg: Signature made Sun Jun 29 11:51:17 2025 -03
gpg:                using RSA key 36006F0E0BA167832158821138EEACA1AB8A6E72
gpg:                issuer "security@coreruleset.org"
gpg: Good signature from "OWASP Core Rule Set <security@coreruleset.org>" [ultimate]
Primary key fingerprint: 3600 6F0E 0BA1 6783 2158  8211 38EE ACA1 AB8A 6E72
```

**Comment by sergioprg** [comment]
Hi, Felipe! Thank you for your work on CRS and for clarifying the mismatch in release file naming in v4.16.0.

I'm maintaining an automated script that validates the `.tar.gz` file using its corresponding `.asc` GPG signature. Until version 4.15.0, the naming convention was consistent (`coreruleset-x.y.z.tar.gz` + `coreruleset-x.y.z.tar.gz.asc`), which allowed secure and fully automated signature verification.

In v4.16.0, the absence of the `coreruleset-4.16.0.tar.gz` file —despite the presence of its `.asc` signature— breaks this logic and introduces ambiguity in automation and security checks.

### Question:
Will future releases return to the consistent naming used in earlier versions (4.15.0 and before)? Or should we now consider the `vX.Y.Z.tar.gz` naming pattern to be the new standard and adapt accordingly?

I assumed the previous naming pattern would be maintained, and I didn't notice the change before implementing the new logic — apologies for not catching it earlier.

Understanding your intended naming policy would help us structure automation scripts logically and securely, avoiding mismatches or signature validation failures.

Thanks again for your time and excellent work!

**Comment by sergioprg** [comment]
## Question regarding signature validation and release file naming conventions for OWASP CRS

Hi Felipe!,

First of all, I’d like to apologize if there was any confusion earlier. We had modified our automation script using AI assistance, and that introduced an unintended change in the GPG verification logic — which led to some incorrect assumptions during testing. I appreciate your previous clarifications and your time.

After adjusting the script, I tested it again using the exact URLs and GPG key you provided — and everything worked correctly. The signature was verified as expected, and the automation completed without issues. Thanks again for helping us get it right.

Now, just to close this topic and for the sake of documentation (and for others automating secure CRS updates), I’d like to confirm a couple of final points.

---

### 🔐 GPG signature validation

Our script performs strict validation using the full OpenPGP fingerprint `36006F0E0BA167832158821138EEACA1AB8A6E72` (associated with `security@coreruleset.org`). It checks both the cryptographic validity of the `.asc` signature and that the signer exactly matches this fingerprint.

**Can we safely assume this key will remain the official and exclusive signing key for future CRS releases?**  
If not, is there a recommended way to detect key rotation automatically, or should we manually update our logic whenever a new key is introduced?

---

### 📦 Release file naming conventions

We are currently using this pattern:

- Source code: `https://github.com/coreruleset/coreruleset/archive/refs/tags/vX.Y.Z.tar.gz`
- Signature: `https://github.com/coreruleset/coreruleset/releases/download/vX.Y.Z/coreruleset-X.Y.Z.tar.gz.asc`

**Can we rely on this file naming convention being stable in future releases?**  
A change in naming or location would break our automation unless there's a programmatic way to determine the correct signature and file to match.

---

Thanks again for your time and for the great work you do on the project. These confirmations will help ensure a robust and secure integration for our environments and hopefully benefit others as well.

**Comment by theseion** [comment]
Hi @sergioprg. 

> Can we safely assume this key will remain the official and exclusive signing key for future CRS releases?

Yes. We intend to continue using this key. Unless we lose access to the key, or it is stolen, we will not need to change it.

> Can we rely on this file naming convention being stable in future releases?

Yes. I don't see any reason that we would change these URLs in the foreseeable future. The tagging convention is also stable.
